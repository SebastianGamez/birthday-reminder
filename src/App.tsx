// Import hooks
import { useState } from 'react';

// Import interfaces
import { IFriendsContext } from "./interfaces/FriendsContextState";

// Import friends
import friends from './data/friends';

// Import context
import FriendsContext from './contexts/FriendsContext';

// Import components
import { ListFriends } from './components/ListFriends';
import { Button } from './components/Button';

// It returns the application
function App(): JSX.Element {

  // Create the state to save in the global context (friendsContext)
  const [friendsContextState, setFriendsContextState] = useState<IFriendsContext>(friends);

  // Create handle of click to clear the list
  const handleClick = (): void => setFriendsContextState({friends: []});
  
  return (
    <main className="root__main" >
      <div className="main__reminder" >
          {/** The title has the list length count */}
          <h1 className="reminder__title" >{friendsContextState.friends.length} birthdays today</h1>
          
          {/** Sharing the state with the components that require it */}
          <FriendsContext.Provider value={ {friendsContextState, setFriendsContextState} }>
            
            {/** Render the list of friends*/}
            <ListFriends />

          </FriendsContext.Provider>

          {/** It button clears the list */}
          <Button handleClick={ handleClick } value="Clear all" labelText="clear" />
      </div>
    </main>
  );
}

export default App;
