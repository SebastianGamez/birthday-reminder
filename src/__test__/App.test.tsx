import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App  from "../App";

describe("<App />", (): void => {

    it("There's taking the App snapshot", (): void => {

        render( <App /> );

        expect(screen.getByRole("main")).toMatchSnapshot();

    });

    it("There's testing the clear all button", (): void => {

        render( <App /> );

        const btn: HTMLButtonElement = screen.getByLabelText("clear");

        userEvent.click( btn );

        expect(screen.getByText("0 birthdays today")).toBeInTheDocument();

    });

});