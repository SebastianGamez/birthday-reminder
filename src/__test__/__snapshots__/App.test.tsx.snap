// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> There's taking the App snapshot 1`] = `
<main
  class="root__main"
>
  <div
    class="main__reminder"
  >
    <h1
      class="reminder__title"
    >
      5
       birthdays today
    </h1>
    <ul
      aria-label="list_friends"
      class="reminder__list"
    >
      <li
        aria-label="friend"
        class="list__friend"
      >
        <div
          class="friend__image"
        >
          <img
            alt="Bertie Yates"
            class="image__image--friend"
            src="https://res.cloudinary.com/diqqf3eq2/image/upload/v1595959131/person-2_ipcjws.jpg"
          />
        </div>
        <div
          class="friend__information"
        >
          <h3
            class="information__name--friend"
          >
            Bertie Yates
          </h3>
          <p
            class="information__age--friend"
          >
            29
             years
          </p>
        </div>
      </li>
      <li
        aria-label="friend"
        class="list__friend"
      >
        <div
          class="friend__image"
        >
          <img
            alt="Hester Hogan"
            class="image__image--friend"
            src="https://res.cloudinary.com/diqqf3eq2/image/upload/v1595959131/person-3_rxtqvi.jpg"
          />
        </div>
        <div
          class="friend__information"
        >
          <h3
            class="information__name--friend"
          >
            Hester Hogan
          </h3>
          <p
            class="information__age--friend"
          >
            32
             years
          </p>
        </div>
      </li>
      <li
        aria-label="friend"
        class="list__friend"
      >
        <div
          class="friend__image"
        >
          <img
            alt="Larry Little"
            class="image__image--friend"
            src="https://res.cloudinary.com/diqqf3eq2/image/upload/v1586883423/person-4_t9nxjt.jpg"
          />
        </div>
        <div
          class="friend__information"
        >
          <h3
            class="information__name--friend"
          >
            Larry Little
          </h3>
          <p
            class="information__age--friend"
          >
            36
             years
          </p>
        </div>
      </li>
      <li
        aria-label="friend"
        class="list__friend"
      >
        <div
          class="friend__image"
        >
          <img
            alt="Sean Walsh"
            class="image__image--friend"
            src="https://res.cloudinary.com/diqqf3eq2/image/upload/v1586883417/person-3_ipa0mj.jpg"
          />
        </div>
        <div
          class="friend__information"
        >
          <h3
            class="information__name--friend"
          >
            Sean Walsh
          </h3>
          <p
            class="information__age--friend"
          >
            34
             years
          </p>
        </div>
      </li>
      <li
        aria-label="friend"
        class="list__friend"
      >
        <div
          class="friend__image"
        >
          <img
            alt="Lola Gardner"
            class="image__image--friend"
            src="https://res.cloudinary.com/diqqf3eq2/image/upload/v1586883334/person-1_rfzshl.jpg"
          />
        </div>
        <div
          class="friend__information"
        >
          <h3
            class="information__name--friend"
          >
            Lola Gardner
          </h3>
          <p
            class="information__age--friend"
          >
            29
             years
          </p>
        </div>
      </li>
    </ul>
    <button
      aria-label="clear"
      class="button__button"
    >
      Clear all
    </button>
  </div>
</main>
`;
